index.ts

```diff
@@ -0,0 +1,50 @@
+import express, { Request, Response } from 'express';
+import { calculateBmi } from './src/bmiCalculator';
+import { calculateExercises } from './src/exerciseCalculator';
+
+const app = express();
+app.use(express.json());
+const PORT = 3003;
+
+app.get('/hello', (_req: Request, res: Response): void => {
+  res.send('Hello Full Stack!');
+});
+
+app.get('/bmi', (req: Request, res: Response): void => {
+  const height = Number(req.query.height);
+  const weight = Number(req.query.weight);
+
+  if (isNaN(height) || isNaN(weight) || height <= 0 || weight <= 0) {
+    res.status(400).json({ error: 'malformatted parameters' });
+    return;
+  }
+
+  const bmi = calculateBmi(height, weight);
+  res.json({ weight, height, bmi });
+});
+
+app.post('/exercises', (req, res): void => {
+  const { daily_exercises, target } = req.body;
+
+  if (!daily_exercises || target === undefined) {
+    res.status(400).json({ error: 'parameters missing' });
+    return;
+  }
+
+  if (
+    !Array.isArray(daily_exercises) ||
```

Comment:
I would separate these checks, one for if it's an array, one for if the array only consists of numbers and one that checks if target is a number.

With that being said - data validation is best left to using a library. I'm using `zod` (https://zod.dev/) I would recommend you try changing this validation to use `zod` instead. Here's an article talking about using a middleware to validate the data when using express: https://dev.to/osalumense/validating-request-data-in-expressjs-using-zod-a-comprehensive-guide-3a0j